sematic/ui/packages/common/src/pages/RunDetails/metrics/Metrics.tsx

```diff
@@ -33,42 +34,50 @@ export function TimeseriesMetric(props: {
     const { metricsFilter, color, latestMetricPoints } = props;
     const [payload, , error] = useMetrics(metricsFilter, latestMetricPoints);
 
-    const [chartData, setChartData] = useState<ChartData<"line", number[], string>>({
-        labels: [],
-        datasets: [
-            {
-                label: metricsFilter.metricName,
-                data: [],
-                backgroundColor: color,
-                borderColor: color,
-            },
-        ],
+    const dataStore = useRef({
```

Comment:
The existing data will be stored in the `useRef()` until it's updated next time.